src/transformers/custom-elements-manifest/get-declarations.ts

```diff
@@ -0,0 +1,334 @@
+import * as tsModule from "typescript";
+import { isSimpleType, toSimpleType } from "ts-simple-type";
+import { ComponentDeclaration } from "../../analyze/types/component-declaration";
+import { getMixinHeritageClauses, getSuperclassHeritageClause } from "../../analyze/util/component-declaration-util";
+import { findParent, getNodeName } from "../../analyze/util/ast-util";
+import { getJsDoc } from "../../analyze/util/js-doc-util";
+import { getTypeHintFromType } from "../../util/get-type-hint-from-type";
+import { AnalyzerResult } from "../../analyze/types/analyzer-result";
+import * as schema from "./schema";
+import { TransformerContext } from "../transformer-context";
+import {
+	typeToSchemaType,
+	getSummaryFromJsDoc,
+	getParameterFromJsDoc,
+	getReturnFromJsDoc,
+	getReferenceFromHeritageClause,
+	getInheritedFromReference
+} from "./utils";
+
+/**
+ * Returns variables in an analyzer result
+ * @param result
+ * @param context
+ */
+function* getExportedDeclarationsFromResult(result: AnalyzerResult, context: TransformerContext): IterableIterator<schema.Declaration> {
+	// Get all export symbols in the source file
+	const symbol = context.checker.getSymbolAtLocation(result.sourceFile);
+	if (symbol == null) {
+		return;
+	}
+
+	const exports = context.checker.getExportsOfModule(symbol);
+
+	// Convert all export variables to VariableDocs
+	for (const exp of exports) {
+		const node = exp.valueDeclaration;
+
+		switch (exp.flags) {
+			case tsModule.SymbolFlags.BlockScopedVariable:
+			case tsModule.SymbolFlags.Variable: {
+				if (tsModule.isVariableDeclaration(node)) {
+					// Get the nearest variable statement in order to read the jsdoc
+					const variableStatement = findParent(node, tsModule.isVariableStatement) || node;
+					const jsDoc = getJsDoc(variableStatement, tsModule);
+
+					yield {
+						kind: "variable",
+						name: node.name.getText(),
+						description: jsDoc?.description,
+						type: typeToSchemaType(context, context.checker.getTypeAtLocation(node)),
+						summary: getSummaryFromJsDoc(jsDoc)
+					};
+				}
+				break;
+			}
+			case tsModule.SymbolFlags.Function: {
+				if (tsModule.isFunctionDeclaration(node) && node.name) {
+					const jsDoc = getJsDoc(node, tsModule);
+					const parameters: schema.Parameter[] = [];
+					let returnType: tsModule.Type | undefined = undefined;
+
+					for (const param of node.parameters) {
+						const name = param.name.getText();
+						const { description, typeHint } = getParameterFromJsDoc(name, jsDoc);
+						const type = typeToSchemaType(context, typeHint || (param.type != null ? context.checker.getTypeAtLocation(param.type) : undefined));
```

Comment:
Run the formatter for the long line?

Reply:
the formatting in this repo seems way off. i think one is setup but badly configured because it seems to do nothing...

ill add a prettierrc and see how much of a change it causes